### PR TITLE
add gitc co [LH ticket #]

### PR DESCRIPTION
**Pull request by richardking**
Implementing 'gitc co [LH ticket #]'

### CODE REVIEW

File: lib/gitcycle.rb [hunk 1]
```diff
@@ -128,6 +128,25 @@ def checkout(*args)
         puts "\nBranch not found!\n".red
         puts "\nDid you mean: gitc branch #{args[0]}\n".yellow
       end
+    elsif args[0] =~ /^\d*$/
+      puts "\nLooking for a branch for LH ticket ##{args[0]}.\n".green
+      results_num = run("git branch | grep \"\\-#{args[0]}\\-\"", :catch => false).scan(/\n/).count
```
Comment (winton) [inline lib/gitcycle.rb]: Would it be cleaner to use the `branches` method here, and then filter the array of branches in Ruby?

Comment (richardking) [inline lib/gitcycle.rb]: I think `branches` returns a string, unless i'm using it wrong? but i did end up using it to clean up one of the calls in the code below.

Comment (winton) [inline lib/gitcycle.rb]: You're right, let's do something like this:

``` ruby
branches.split(/\*\s/).detect { |b| b.include?("-#{args[0]}-") }
```

Comment (winton) [inline lib/gitcycle.rb]: You could also add an `:array => true` option to the branches method for future use.

**Comment by richardking** [comment]
thanks for the tip. made the updates and looks way better now i think.

here's what the different outputs look like for your reference as well:
http://d.pr/i/jWCD

**Comment by winton** [comment]
You're a baller, Richard, thanks!

**Pull request merged**